AEPMessaging/Sources/MessagingPropositionItem.swift

```diff
@@ -56,8 +56,11 @@ public class MessagingPropositionItem: NSObject, Codable {
 
         itemId = try container.decode(String.self, forKey: .id)
         schema = try SchemaType(from: container.decode(String.self, forKey: .schema))
-        let codableItemData = try? container.decode([String: AnyCodable].self, forKey: .data)
-        itemData = AnyCodable.toAnyDictionary(dictionary: codableItemData)
+        let codableItemData = try container.decode([String: AnyCodable].self, forKey: .data)
+        itemData = AnyCodable.toAnyDictionary(dictionary: codableItemData) ?? [:]
+        guard !itemData.isEmpty else {
+            throw DecodingError.dataCorruptedError(forKey: CodingKeys.data, in: container, debugDescription: "Item data is corrupted and cannot be decoded.")
```

Comment:
debug description is a bit misleading here i think - the data isn't corrupt, it's just empty which is unexpected.

Reply:
As discussed offline, I've updated the implementation to allow empty data for holdout population. Also added a test for the same.